pkg/gpu/nvidia/metrics/util.go

```diff
@@ -83,12 +98,16 @@ func AverageGPUUtilization(uuid string, since time.Duration) (uint, error) {
 	lastTs := C.ulonglong(time.Now().Add(-1*since).UnixNano() / 1000)
 	uuidCStr := C.CString(uuid)
 	var util C.uint
-	r := C.nvmlDeviceGetAverageUsage(uuidCStr, lastTs, &util)
+	var sum C.uint
+	var sampleCount C.uint
+	r := C.nvmlDeviceGetAverageUsage(uuidCStr, lastTs, &util, &sum, &sampleCount)
 	C.free(unsafe.Pointer(uuidCStr))
 	if r != C.NVML_SUCCESS {
 		return 0, fmt.Errorf("failed to get GPU utilization for device %s, nvml return code: %v", uuid, r)
-
 	}
-
-	return uint(util), nil
+	averageUtilization := uint(util)
+	if averageUtilization > 100 || averageUtilization < 0 {
+		return 0, fmt.Errorf("failed to get GPU utilization for device %s, out of range [0, 100] utilization: %d, sum: %d, sampleCount: %d", uuid, averageUtilization, uint(sum), uint(sampleCount))
```

Comment:
How will the error be handled from the function calller?

Reply:
It will report empty metric, and log error in the device plugin: https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/master/pkg/gpu/nvidia/metrics/metrics.go#L196-L199.